### ISSUE TITLE
Variable "e" was not defined 

### ISSUE THREAD

**Issue by GENGSHUANGs** [javascript-obfuscator/javascript-obfuscator#321]
Example code :

```javascript
function e() {
    function e() {
    }
    e();
}

n.init(function() {
    e();
});
```

Obfuscatored code is (I am using obfuscator.io):

```javascript
var _0x1744 = ['aW5pdA=='];
(function (_0x448006, _0x1654c2) {
    var _0x52c3a5 = function (_0x5b8784) {
        while (--_0x5b8784) {
            _0x448006['push'](_0x448006['shift']());
        }
    };
    _0x52c3a5(++_0x1654c2);
}(_0x1744, 0x1bd));
var _0x4a80 = function (_0x252da1, _0x2e38a8) {
    _0x252da1 = _0x252da1 - 0x0;
    var _0x3ec7f1 = _0x1744[_0x252da1];
    if (_0x4a80['NDtGsL'] === undefined) {
        (function () {
            var _0x4122e8 = function () {
                var _0x10d66d;
                try {
                    _0x10d66d = Function('return\x20(function()\x20' + '{}.constructor(\x22return\x20this\x22)(\x20)' + ');')();
                } catch (_0x32087a) {
                    _0x10d66d = window;
                }
                return _0x10d66d;
            };
            var _0xc2b41b = _0x4122e8();
            var _0x3ba9f8 = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=';
            _0xc2b41b['atob'] || (_0xc2b41b['atob'] = function (_0xb0b7e3) {
                var _0x239f05 = String(_0xb0b7e3)['replace'](/=+$/, '');
                for (var _0x33c4d7 = 0x0, _0x47e838, _0x4b87e5, _0x54f9ae = 0x0, _0x526468 = ''; _0x4b87e5 = _0x239f05['charAt'](_0x54f9ae++); ~_0x4b87e5 && (_0x47e838 = _0x33c4d7 % 0x4 ? _0x47e838 * 0x40 + _0x4b87e5 : _0x4b87e5, _0x33c4d7++ % 0x4) ? _0x526468 += String['fromCharCode'](0xff & _0x47e838 >> (-0x2 * _0x33c4d7 & 0x6)) : 0x0) {
                    _0x4b87e5 = _0x3ba9f8['indexOf'](_0x4b87e5);
                }
                return _0x526468;
            });
        }());
        _0x4a80['WbnAJW'] = function (_0x206ee2) {
            var _0x9e7010 = atob(_0x206ee2);
            var _0x1f6913 = [];
            for (var _0x36e106 = 0x0, _0x2466d2 = _0x9e7010['length']; _0x36e106 < _0x2466d2; _0x36e106++) {
                _0x1f6913 += '%' + ('00' + _0x9e7010['charCodeAt'](_0x36e106)['toString'](0x10))['slice'](-0x2);
            }
            return decodeURIComponent(_0x1f6913);
        };
        _0x4a80['VmGoxB'] = {};
        _0x4a80['NDtGsL'] = !![];
    }
    var _0x5e30a2 = _0x4a80['VmGoxB'][_0x252da1];
    if (_0x5e30a2 === undefined) {
        _0x3ec7f1 = _0x4a80['WbnAJW'](_0x3ec7f1);
        _0x4a80['VmGoxB'][_0x252da1] = _0x3ec7f1;
    } else {
        _0x3ec7f1 = _0x5e30a2;
    }
    return _0x3ec7f1;
};
function _0x46b8d6() {
    function _0x46b8d6() {
    }
    _0x46b8d6();
}
n[_0x4a80('0x0')](function () {
    e(); // <---------------------------------- e was not defined 
});
```

**Comment by GENGSHUANGs** [comment]
Example code:
```javascript
function e() {
    function e() {
    }
    e();
}

var n = {
	init:function(fn){
		fn && fn();
	}
};

n.init(function() {
    e();
});
```


Obfuscated code is :
```javascript
var _0xc5e1 = ['aW5pdA=='];
(function (_0xcac8b8, _0x4f9566) {
    var _0x34de08 = function (_0x18d511) {
        while (--_0x18d511) {
            _0xcac8b8['push'](_0xcac8b8['shift']());
        }
    };
    _0x34de08(++_0x4f9566);
}(_0xc5e1, 0x1cb));
var _0x5c0c = function (_0xfbd285, _0xd9344) {
    _0xfbd285 = _0xfbd285 - 0x0;
    var _0x2b84ab = _0xc5e1[_0xfbd285];
    if (_0x5c0c['ksSjoC'] === undefined) {
        (function () {
            var _0x4563b4 = function () {
                var _0x3b8f0d;
                try {
                    _0x3b8f0d = Function('return\x20(function()\x20' + '{}.constructor(\x22return\x20this\x22)(\x20)' + ');')();
                } catch (_0x5ee1e3) {
                    _0x3b8f0d = window;
                }
                return _0x3b8f0d;
            };
            var _0x964c1e = _0x4563b4();
            var _0x46e6c8 = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=';
            _0x964c1e['atob'] || (_0x964c1e['atob'] = function (_0x19ac3f) {
                var _0x59b98d = String(_0x19ac3f)['replace'](/=+$/, '');
                for (var _0x57d4e8 = 0x0, _0xd9af73, _0x1a9a74, _0x44b6de = 0x0, _0x5f07f7 = ''; _0x1a9a74 = _0x59b98d['charAt'](_0x44b6de++); ~_0x1a9a74 && (_0xd9af73 = _0x57d4e8 % 0x4 ? _0xd9af73 * 0x40 + _0x1a9a74 : _0x1a9a74, _0x57d4e8++ % 0x4) ? _0x5f07f7 += String['fromCharCode'](0xff & _0xd9af73 >> (-0x2 * _0x57d4e8 & 0x6)) : 0x0) {
                    _0x1a9a74 = _0x46e6c8['indexOf'](_0x1a9a74);
                }
                return _0x5f07f7;
            });
        }());
        _0x5c0c['tmWLZx'] = function (_0x2ee29c) {
            var _0x506f21 = atob(_0x2ee29c);
            var _0x3c5070 = [];
            for (var _0x744a02 = 0x0, _0x12b3e3 = _0x506f21['length']; _0x744a02 < _0x12b3e3; _0x744a02++) {
                _0x3c5070 += '%' + ('00' + _0x506f21['charCodeAt'](_0x744a02)['toString'](0x10))['slice'](-0x2);
            }
            return decodeURIComponent(_0x3c5070);
        };
        _0x5c0c['lbYhdQ'] = {};
        _0x5c0c['ksSjoC'] = !![];
    }
    var _0x6712ad = _0x5c0c['lbYhdQ'][_0xfbd285];
    if (_0x6712ad === undefined) {
        _0x2b84ab = _0x5c0c['tmWLZx'](_0x2b84ab);
        _0x5c0c['lbYhdQ'][_0xfbd285] = _0x2b84ab;
    } else {
        _0x2b84ab = _0x6712ad;
    }
    return _0x2b84ab;
};
function _0x5e4ced() {
    function _0x5e4ced() {
    }
    _0x5e4ced();
}
var n = {
    'init': function (_0x5cf327) {
        _0x5cf327 && _0x5cf327();
    }
};
n[_0x5c0c('0x0')](function () {
    e(); // <--------------- Here the variable must be "_0x5e4ced"
});

```

**Comment by sanex3339** [comment]
Hi, i'll look into this issue but currently i take a 1-2 months timeout with this project to doing renovations on my flat. So, until october i'll have no time to look into this.

This is high priority issue, so if i'll have time, i look into this.

**Comment by beeing** [comment]
Hi, any updates on this? I'm having similar error as well

**Comment by zamotkin** [comment]
Fixed by #440

**Comment by sanex3339** [comment]
Released in 0.18.5
